src/bls12_381/ec.rs

```diff
@@ -174,6 +184,33 @@ macro_rules! curve_impl {
             }
         }
 
+        impl $affine {
+
+            fn y2_from_x(x: $basefield) -> $basefield {
+                let mut y2 = x.clone();
+                y2.square();
+                y2.mul_assign(&x);
+                y2.add_assign(&Self::get_coeff_b());
+                y2
+            }
+
+            fn is_on_curve(&self) -> bool {
+                if self.is_zero() {
+                    true
+                } else {
+                    // Check that the point is on the curve
+                    let mut y2 = self.y;
+                    y2.square();
+
+                    y2 == $affine::y2_from_x(self.x)
+                }
+            }
+
+            fn is_in_correct_subgroup(&self) -> bool {
```

Comment:
Oh right, I see now looking at the source code, and indeed it makes sense to have it separatly: see src/bls12_381/ec.rs:1379 or src/bls12_381/ec.rs:785.
Since this naming change does not directly concern this pull request, I'll open a new one.